src/capi/typeobject.cpp

```diff
@@ -1729,7 +1729,9 @@ extern "C" int PyType_Ready(PyTypeObject* cls) noexcept {
 
     RELEASE_ASSERT(cls->tp_getattro == NULL || cls->tp_getattro == PyObject_GenericGetAttr, "");
     RELEASE_ASSERT(cls->tp_setattro == NULL || cls->tp_setattro == PyObject_GenericSetAttr, "");
-    RELEASE_ASSERT(cls->tp_as_buffer == NULL, "");
+
+    // TODO: I think this is not implemented yet but the arraymodules uses it.
+    // RELEASE_ASSERT(cls->tp_as_buffer == NULL, "");
```

Comment:
These asserts are in here for a reason :P  We should definitely prefer failing loudly over quietly doing the wrong thing.

I think in this particular case, tp_as_buffer is ok to ignore, since its uses are pretty limited and we don't have a Pyston analogue that it conflicts with.  Could you verify that and then remove this assert entirely?